gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/RequestHandler.java

```diff
@@ -107,9 +111,19 @@ public ExploreResponse.Builder handleRequest(
         buildQueryRequest(requestContext, request, attributeMetadataProvider);
 
     Iterator<ResultSetChunk> resultSetChunkIterator = executeQuery(requestContext, queryRequest);
+    ExploreResponse.Builder responseBuilder =
+        handleQueryServiceResponse(
+            request,
+            requestContext,
+            resultSetChunkIterator,
+            requestContext,
+            attributeMetadataProvider);
+
+    if (request.getFetchTotal()) {
```

Comment:
Doing this in sequence is going to double-up the latency.
I guess, it's fine as long as the total latency is acceptable.
If required, can explore parallelising this.

Reply:
Yeah lets try with sequential for now. I am planning for the UI to query total separately then the data call